src/carnot/funcs/builtins/pprof_ops.h

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2018- The Pixie Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <sys/sysinfo.h>
+
+#include <absl/container/flat_hash_map.h>
+
+#include <string>
+
+#include "src/shared/pprof/pprof.h"
+
+#include "src/carnot/udf/registry.h"
+#include "src/carnot/udf/udf.h"
+
+namespace px {
+namespace carnot {
+namespace builtins {
+
+using px::shared::PProfProfile;
+
+class CreatePProfRowAggregate : public udf::UDA {
+ public:
+  static udf::UDADocBuilder Doc() {
+    return udf::UDADocBuilder("Convert perf profiling data to pprof format.")
+        .Details("Converts perf profiling stack traces into pprof format.")
+        .Example(
+            R"example(df = px.DataFrame(table='stack_traces.beta', start_time='-1m')
+df = df.agg(pprof=('stack_trace', 'count', px.pprof)))example")
+        .Arg("stack_trace", "Stack trace string.")
+        .Arg("count", "Count of the stack trace string.")
+        .Returns("A single row that aggregates all the stack traces and counts into pprof format.");
+  }
+
+  void Update(FunctionContext*, const StringValue stack_trace, const Int64Value count) {
+    histo_[stack_trace] += count.val;
+  }
+
+  void Merge(FunctionContext*, const CreatePProfRowAggregate& other) {
+    for (const auto& [stack_trace, count] : other.histo_) {
+      histo_[stack_trace] += count;
+    }
+  }
+
+  StringValue Serialize(FunctionContext*) {
+    static constexpr uint32_t stack_trace_period_ms = 11;
```

Comment:
I assume this is expected to line up with the flag `stirling_profiler_stack_trace_sample_period_ms`?
Seems strange and incorrect to hard code it here. What happens if someone overrides the flag?

Reply:
Agreed, there is a potential mismatch. We added issue #1352 for this. Curious to get feedback, should we:
1. Pursue #1352 first (and mark this as draft) or,
2. Just land this and triage #1352 instead?
i.e. because although it is possible for the parameters to be out-of-sync we don't think this is a common setup (and since the use case for export isn't fully end-to-end setup yet, we have time also).

Reply:
IMO it's a bit strange to land this PR first. Even though something is unlikely to happen, we should avoid leaving the codebase in a state where one could encounter unexpected behavior.
My vote would be option 1.

Reply:
Converted to draft. Will pursue #1352 first.